### ISSUE TITLE
ppc64: audio glitch

### ISSUE THREAD

**Issue by MingcongBai** [mgba-emu/mgba#460]
I am currently running a custom build of mGBA 0.5.1 on a PowerPC G5 system:

```
jeffbai@G5-AOSC [ ~ ] $ LANG=C lscpu
Architecture:          ppc64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Big Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    1
Core(s) per socket:    1
Socket(s):             4
NUMA node(s):          1
Model:                 1.1 (pvr 0044 0101)
Model name:            PPC970MP, altivec supported
CPU max MHz:           2500.0000
CPU min MHz:           1250.0000
L1d cache:             32K
L1i cache:             64K
L2 cache:              1024K
NUMA node0 CPU(s):     0-3
```

Where while the games played fine, the emulator only seemed to play white noise. The audio settings only showed a "Qt Multimedia" option for audio output.

What other information might be useful for debugging this issue? Please let me know.

**Comment by endrift** [comment]
Try this patch and let me know if it works:

```diff
diff --git a/src/platform/qt/AudioProcessorQt.cpp b/src/platform/qt/AudioProcessorQt.cpp
index 90fca28..b2b6ee4 100644
--- a/src/platform/qt/AudioProcessorQt.cpp
+++ b/src/platform/qt/AudioProcessorQt.cpp
@@ -50,7 +50,7 @@ bool AudioProcessorQt::start() {
 		format.setChannelCount(2);
 		format.setSampleSize(16);
 		format.setCodec("audio/pcm");
-		format.setByteOrder(QAudioFormat::LittleEndian);
+		format.setByteOrder(QSysInfo::ByteOrder);
 		format.setSampleType(QAudioFormat::SignedInt);
 
 		m_audioOutput = new QAudioOutput(format, this);
```

**Comment by MingcongBai** [comment]
I did rebase a patch for 0.5.1:

```patch
diff -Naur mgba-0.5.1/src/platform/qt/AudioProcessorQt.cpp mgba-0.5.1.endian/src/platform/qt/AudioProcessorQt.cpp
--- mgba-0.5.1/src/platform/qt/AudioProcessorQt.cpp	2016-10-05 15:39:41.000000000 -0500
+++ mgba-0.5.1.endian/src/platform/qt/AudioProcessorQt.cpp	2016-11-06 16:29:53.428981899 -0600
@@ -50,7 +50,7 @@
 		format.setChannelCount(2);
 		format.setSampleSize(16);
 		format.setCodec("audio/pcm");
-		format.setByteOrder(QAudioFormat::LittleEndian);
+		format.setByteOrder(QSysInfo::ByteOrder);
 		format.setSampleType(QAudioFormat::SignedInt);
 
 		m_audioOutput = new QAudioOutput(format, this);
```

However, I got an error when compiling after the patch was applied:

```
/var/lib/abbs/build/tmp.leLSvRwCN7/mgba-0.5.1/src/platform/qt/AudioProcessorQt.cpp: In member function 'virtual bool QGBA::AudioProcessorQt::start()':
/var/lib/abbs/build/tmp.leLSvRwCN7/mgba-0.5.1/src/platform/qt/AudioProcessorQt.cpp:53:42: error: no matching function for call to 'QAudioFormat::setByteOrder(QSysInfo::Endian)'
   format.setByteOrder(QSysInfo::ByteOrder);
                                          ^
In file included from /usr/include/qt5/QtMultimedia/QAudioFormat:1:0,
                 from /var/lib/abbs/build/tmp.leLSvRwCN7/mgba-0.5.1/src/platform/qt/AudioDevice.h:9,
                 from /var/lib/abbs/build/tmp.leLSvRwCN7/mgba-0.5.1/src/platform/qt/AudioProcessorQt.cpp:8:
/usr/include/qt5/QtMultimedia/qaudioformat.h:82:10: note: candidate: void QAudioFormat::setByteOrder(QAudioFormat::Endian)
     void setByteOrder(QAudioFormat::Endian byteOrder);
          ^~~~~~~~~~~~
/usr/include/qt5/QtMultimedia/qaudioformat.h:82:10: note:   no known conversion for argument 1 from 'QSysInfo::Endian' to 'QAudioFormat::Endian'
make[2]: *** [qt/CMakeFiles/mgba-qt.dir/build.make:1326: qt/CMakeFiles/mgba-qt.dir/AudioProcessorQt.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:162: qt/CMakeFiles/mgba-qt.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
```

Also, I could build mGBA on ppc64 when SDL backend was enabled, I will open a separate issue for it.

**Comment by MingcongBai** [comment]
A working patch, thanks to @LER0ever.

```patch
diff -Naur mgba-0.5.1/src/platform/qt/AudioProcessorQt.cpp mgba-0.5.1.endian/src/platform/qt/AudioProcessorQt.cpp
--- mgba-0.5.1/src/platform/qt/AudioProcessorQt.cpp	2016-10-05 15:39:41.000000000 -0500
+++ mgba-0.5.1.endian/src/platform/qt/AudioProcessorQt.cpp	2016-11-06 16:49:01.988825308 -0600
@@ -50,7 +50,7 @@
 		format.setChannelCount(2);
 		format.setSampleSize(16);
 		format.setCodec("audio/pcm");
-		format.setByteOrder(QAudioFormat::LittleEndian);
+		format.setByteOrder(QAudioFormat::Endian(QSysInfo::ByteOrder));
 		format.setSampleType(QAudioFormat::SignedInt);
 
 		m_audioOutput = new QAudioOutput(format, this);
```

**Comment by endrift** [comment]
Fixed in 60974cf61b0da1056f19d1b2b2127017736186d8.

**Comment by MingcongBai** [comment]
Good work, thanks.
